rfcs/2022-10-31-13691-tooling-revamp.md

```diff
@@ -0,0 +1,209 @@
+# RFC 15056 - 2022-10-31 - Tooling revamp
+
+-----
+
+This RFC discusses improving Vector's developer tooling in order to ease maintenance and expand testing possibilities.
+
+**Table of Contents**
+
+- [Scope](#scope)
+  - [In scope](#in-scope)
+  - [Out of scope](#out-of-scope)
+- [Pain](#pain)
+- [Proposal](#proposal)
+  - [User Experience](#user-experience)
+  - [Implementation](#implementation)
+    - [Environment management](#environment-management)
+    - [Configuration](#configuration)
+    - [Interface](#interface)
+    - [Testing container](#testing-container)
+    - [CI](#ci)
+- [Drawbacks](#drawbacks)
+- [Prior Art](#prior-art)
+- [Alternatives](#alternatives)
+  - [Status quo](#status-quo)
+  - [Python](#python)
+- [Outstanding Questions](#outstanding-questions)
+- [Plan Of Attack](#plan-of-attack)
+- [Future Improvements](#future-improvements)
+
+## Scope
+
+### In scope
+
+- Begin the standardization of interfacing with the repository (building, testing, releasing, etc.)
+- Define a version support matrix for each integration
+- Only run an integration's tests on PRs for changes
+- Make developer UX consistent across platforms
+
+### Out of scope
+
+- Non-integration tests
+
+## Pain
+
```

Comment:
Makefiles and scripts can get messy fast and often are hard to scale well.

Reply:
Added!